File: test/cases/semantic_valid/sv17_func_result.pas
program sv17_func_result;
var
    n, result: integer;

function factorial(x: integer): integer;
begin
    if x <= 1 then
        factorial := 1
    else
        factorial := x * factorial(x - 1)
end;

begin
    n := 5;
    result := factorial(n)
end.

File: test/cases/semantic_valid/sv17_func_result.c
#include <stdio.h>
int n;
int result;


int factorial(int x);

int factorial(int x) {
    int _retval;
    if ((x <= 1)) {
        _retval = 1;
    } else {
        _retval = (x * factorial((x - 1)));
    }

    return _retval;
}


int main(void) {
    n = 5;
    result = factorial(n);

    return 0;
}
